### PR TITLE
Private beta invite

### PR DESCRIPTION
**Pull request by thomassdk**
## Description

This PR adds the basic capability for private beta invites to the schema models and a lambda to create the invite code.

**New stuff** ✨

1. A model name `PrivateBetaInviteCode` has been added to the schema which keeps tracks of invite codes for the private beta. **Note** the id of each entry is used as the invite code.

2. There is a new lambda called `createPrivateBetaInvite`

The idea is to call this lambda via a lambda function URL but until that is setup to test this locally add this snippet to the end of the root `Mutation` object in the schema:
```graphql
createInviteTest: Boolean
  @function(name: "createPrivateBetaInvite-${env}")
```

The following are the mutations to test the code functionality in the graphql explorer:
```graphql
mutation TestLambda {
  createInviteTest
}

mutation TestMutation {
  createPrivateBetaInviteCode(input: {}) {
    id
    valid
  }
}
```

Some cloudformation code has been added to this lambda to automatically created the function url.

The URL has a parameter called `shareableInvites` which dictates how many invites the code it creates can share.

Once an invitation has been used the `shareableInvites` invites field on the `PrivateBetaInviteCode` model needs to be decremented

**Changes** 🏗

1. A `private` flag has been added to the Colony model 
2. A `colonyMemberInvite` type has also been added to the Colony model that is created whenever a colony is created via the `createUniqueColony` lambda. This type contains the invite code and a count of the number of times this invite code has been used.
3. The colony fragment now has the `private` flag 

### 🏗 Tremendous changes (by Chris) 🏗

The main goal of all the refactoring was to get to a state where we have **all components work also outside the Colony context** (except for the ones that are specific for a Colony of course).

For that I:

- Moved a bunch of stuff around
- Removed dependencies of `useColonyContext()`
- Created new, adjusted components where necessary
- Created nested routes to better resemble the separation of concerns and contexts (these are called `xxxRoute.tsx` in the `routes` directory
- Simplified and streamlined the `Routes.tsx`
- Modified all navigation components and links to point to the right place.
- The user stuff is now under `/my/profile`, `/my/preferences` and `/my/advances-settings`. If we don't like it we can change it back of course

In the process I broke a bunch of things, most notable:
  - Some loading spinners
  - The mobile navigation
  - The three column layout for extension details
  - The Colony Switcher (still works but looks different - this will be changed anyways)
 I will repair all of those in due course

Also, all pages are now embedded in the new navigation layout (old stuff is not yet removed though - like the `<Default>` component.

There's probably a whole bunch of stuff that I forgot to mention. I would really like to merge this though, so that people can work off of this instead of adding clutter to the old system. I'd suggest that we create issues for the stuff that is broken and review this with these issues in mind (and only merge it once all issues are created).

### CODE REVIEW

File: src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx
Comment (ArmandoGraterol) [inline src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx:25]: Why not use the `FormattedMessage` and `Heading` components like we do for the rest of the CDapp?

File: src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx
Comment (ArmandoGraterol) [inline src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx:55]: Same question as above, but I see that in a few places we are using `FormattedMessage` and in some not?

File: src/components/frame/Extensions/pages/ColonyDetailsPage/ColonyDetailsPage.tsx
Comment (ArmandoGraterol) [inline src/components/frame/Extensions/pages/ColonyDetailsPage/ColonyDetailsPage.tsx:34]: We have a component called `ExternalLink` for... external links 😎 👉 👉  I know it was implemented like this already before you touched it (By PagePro perhaps?) but I'm just mentioning it for future reference

File: src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx
Comment (ArmandoGraterol) [inline src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx:188]: There is a lot of repetition in this file, but then again, it was done before you came and updated it and it's not blocking us so I guess the best course of action would be to just take note of it and move on?

File: src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx
Comment (ArmandoGraterol) [inline src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx:15]: We can remove this div and apply the class to the parent `div` if it's still needed

File: src/context/AppContext.tsx
Comment (ArmandoGraterol) [inline src/context/AppContext.tsx:193]: Nice, the previous place where this was looks neater this way

File: src/routes/ColonyRoute.tsx
Comment (ArmandoGraterol) [inline src/routes/ColonyRoute.tsx:27]: Holy

**Review by ArmandoGraterol — COMMENTED** [review]
That's a lot of code! I think it would be best for future reviews if we try to keep PRs below the ~1000 lines of code changed.

I've noticed 2 weird behaviours for now:

- When I go to the colony dashboard, this error appears on the console and the tab title breaks and shows a wrong one "Extensions > ..." 
Screenshot 2023-10-25 at 17 50 52

- When I click the "Create Colony" on the landing page, I get instantly redirected to the landing page, is that intentional? I know we are reworking that flow but shouldn't we display the old one for now?

File: src/components/frame/AvatarDropdown/AvatarDropdownPopoverMobile.tsx
Comment (jakubcolony) [inline src/components/frame/AvatarDropdown/AvatarDropdownPopoverMobile.tsx:24]: As above

File: src/components/frame/Extensions/layouts/ColonyLayout.tsx
Comment (jakubcolony) [inline src/components/frame/Extensions/layouts/ColonyLayout.tsx:7]: And same here

**Review by jakubcolony — APPROVED** [review]
Lots of good work here 💪

File: src/components/frame/Extensions/layouts/Header.tsx
Comment (chmanie) [reply src/components/frame/Extensions/layouts/Header.tsx:3]: Just copied it. But will change.

File: src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx
Comment (chmanie) [reply src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx:15]: Just copied it. But will change.

File: src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx
Comment (chmanie) [reply src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx:188]: Just copied it. But will change.

File: src/components/frame/Extensions/pages/ColonyDetailsPage/ColonyDetailsPage.tsx
Comment (chmanie) [reply src/components/frame/Extensions/pages/ColonyDetailsPage/ColonyDetailsPage.tsx:34]: Just copied it. But will change.

File: src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx
Comment (chmanie) [reply src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx:55]: Just copied it. But will change.

File: src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx
Comment (chmanie) [reply src/components/frame/Extensions/pages/AdvancedPage/AdvancedPage.tsx:25]: Just copied it. But will change.

File: src/components/frame/Extensions/layouts/Header.tsx
Comment (chmanie) [reply src/components/frame/Extensions/layouts/Header.tsx:3]: Just copied it. But will change.

**Comment by chmanie** [comment]
> When I click the "Create Colony" on the landing page, I get instantly redirected to the landing page, is that intentional? I know we are reworking that flow but shouldn't we display the old one for now?

There was a trade off between making this PR even bigger and making this actually functional. I can try to fix this but it might just be much simpler to wait until #1197 is merged. Your call!

> When I go to the colony dashboard, this error appears on the console and the tab title breaks and shows a wrong one

I broke tab titles in this PR (they were inconsistent an kinda broken before as well). I will fix the console error though but the titles I would like to overhaul in a separate PR.

**Pull request merged**
